tests/test_updater_with_simulator.py

```diff
@@ -152,6 +152,33 @@ def test_keys_and_signatures(self):
 
         self._run_refresh()
 
+    def test_snapshot_rollback_check_with_missing_local_snapshot(self):
+        # Test triggering snapshot rollback check on a newly downloaded snapshot
+        # when the local snapshot is not loaded due to hash mismatch.
```

Comment:
the local snapshot does get loaded as intermediate snapshot (as in, we know it won't be valid as final snapshot but we want to load it for the rollback checks)